samples/net/lwm2m_client/src/lwm2m-client.c

```diff
@@ -81,7 +81,11 @@ static int lwm2m_setup(void)
 
 	/* Security Mode */
 	lwm2m_set_u8(&LWM2M_OBJ(0, 0, 2), IS_ENABLED(CONFIG_LWM2M_DTLS_SUPPORT) ? 0 : 3);
+
 #if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+	/* Check if client ID length is valid */
+	BUILD_ASSERT(sizeof(endpoint) <= CONFIG_LWM2M_SECURITY_KEY_SIZE,
```

Comment:
Since it's a build assert, I'd rather see this near the `endpoint` definition, not inside the function.